src/Util/FisherYates/Implementation.dfy

```diff
@@ -0,0 +1,33 @@
+/*******************************************************************************
+ *  Copyright by the contributors to the Dafny Project
+ *  SPDX-License-Identifier: MIT
+ *******************************************************************************/
+
+module FisherYates.Implementation {
+  import Interface
+
+  trait {:termination false} Trait extends Interface.Trait {
+
+    method Shuffle<T(!new)>(a: array<T>)
+      decreases *
```

Comment:
Why decreases *?

Reply:
As discussed offline, the decreases * is inherited from Uniform.